File: sucraft-core/src/main/java/org/sucraft/core/main/SuCraftCorePlugin.java
package org.sucraft.core.main;

import org.sucraft.core.common.plugin.PluginUtils;
import org.sucraft.core.common.plugin.SuCraftPlugin;

public class SuCraftCorePlugin extends SuCraftPlugin {
	
	// Singleton
	
	private static SuCraftCorePlugin instance = null;
	public static SuCraftCorePlugin instance() {
		return PluginUtils.requirePluginInitialized(instance);
	}
	
	// Initialization
	
	@Override
	public void onPluginPreEnable() {
		instance = this;
	}
	
}

File: sucraft-core/src/main/kotlin/org/sucraft/core/main/SuCraftCorePlugin.kt
/*
 * Copyright (c) SuCraft 2021 sucraft.org
 */

package org.sucraft.core.main

import org.sucraft.core.common.sucraft.plugin.SuCraftPlugin
import org.sucraft.core.common.general.pattern.SingletonContainer


class SuCraftCorePlugin : SuCraftPlugin() {

	// Companion (singleton)

	companion object : SingletonContainer<SuCraftCorePlugin>()

}